sintr_worker/lib/basic_info.dart

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2015, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:isolate';
+
+import 'package:gcloud/storage.dart';
+import 'package:sintr_common/logging_utils.dart' as logging;
+import 'package:sintr_common/auth.dart' as auth;
+import 'package:sintr_common/configuration.dart' as config;
+
+var PROJECT = 'sintr-994';
+
+final _log = new logging.Logger("worker_isolate");
+
+main(List<String> args, SendPort sendPort) {
+  logging.setupLogging();
+  _log.fine("args: $args");
+
+  config.configuration = new config.Configuration(PROJECT,
+  cryptoTokensLocation: "${config.userHomePath}/Communications/CryptoTokens");
+
+  ReceivePort receivePort = new ReceivePort();
+  sendPort.send(receivePort.sendPort);
+
+  receivePort.listen((msg) async {
+    sendPort.send(await _protectedHandle(msg));
+  });
+}
+
+Future<String> _protectedHandle(String msg) async {
+  try {
+    // Unpack arguments
+    var inArgs = JSON.decode(msg);
+    _log.finest("inArgs: $msg");
+
+    String key = inArgs["key"];
+    String value = inArgs["value"];
+
+    var response = await map(key, value);
+
+    _log.finest("response: $response");
+    return JSON.encode(response);
+  } catch (e, st) {
+    _log.fine("Execution erred. $e \n $st \n");
+    _log.fine("Input data: $msg");
+    return JSON.encode({});
+  }
+}
+
+Future<Map<String, List<String>>> map(String k, String v) async {
+  var client = await auth.getAuthedClient();
+  _log.finest("Client acquired");
+
+  var storage = new Storage(client, "sintr-994");
+  _log.finest("Storage acquired");
+
+  Stream<String> stream = storage
+      .bucket("sintr-sample-test-data")
+      .read(k)
+      .transform(UTF8.decoder) // Decode bytes to UTF8.
+      .transform(new LineSplitter());
+
+  return await mapStream(k, stream);
+}
+
+const basic_info_key = 'basic_info';
+const skip_value = 'skip';
+const unknown_value = 'unknown';
+const data_prefix = 'Data :: ~';
+
+/// Extract basic information about the client (IDE) being used
+Future<Map<String, List<String>>> mapStream(
+String logKey, Stream<String> stream) async {
+  var result = new Map<String, List<String>>();
+  await for (String line in stream) {
+    if (line.startsWith('sessionID :: ')) {
+      if (!line.startsWith('sessionID :: PRI')) {
+        // The information is only encoded in priority streams
+        return result..[basic_info_key] = [skip_value];
```

Comment:
Adding the cascade means that the map is returned rather than the list of values. If this coding style is difficult to read, I can split it into separate lines... one to assign the value to the map and the second to return the map.

https://dartpad.dartlang.org/d207539ea9a9f7b62062